code/modules/instruments/instrument_data/_instrument_key.dm

```diff
@@ -0,0 +1,20 @@
+/datum/instrument_key
+	var/key			//1 to 127
+	var/sample		//file
+	var/frequency	//frequency generated
+	var/deviation	//deviation up/down towards pivot from sample (??)
+
+/datum/instrument_key/New(sample = src.sample, key = src.key, deviation = src.deviation, frequency = src.frequency)
+	src.sample = sample
+	src.key = key
+	src.deviation = deviation
+	src.frequency = frequency
+	if(!frequency && deviation)
+		calculate()
+
+/datum/instrument_key/proc/calculate()
+	if(!deviation)
+		CRASH("Invalid calculate call: No deviation or sample in instrument_key")
+	#define KEY_TWELTH 0.083
```

Comment:
dont ask why this is named this
i put this in like 2 years ago as a replacement for baystation literally calling the define, and i quote, "Q", so yeah atleasti t's somewhat more descriptive :joy:

Reply:
why is it 0.083 instead of 1/12, which is kinda why it's called "KEY_TWELTH"

Reply:
it's precalculated a 3 decimal place/digit accuracy because if i did (1/12) byond is going to compute 0.08333333333333333333333333333333333 every time something initialized

and while yes, that's uh, technically not too much of an issue it's like y
i'm raising it to a bit higher accuracy tho.

Reply:
fuck it changed it it's only on init